src/components/auth/register/Register.tsx

```diff
@@ -1,48 +1,192 @@
+  confirmPassword: "",
+};
 
-interface onSubmitProps {
-    user: UserProps,
-    link: string,
-    validate: () => boolean;
-}
+// Set yup validation schema to validate defined fields and error messages
+/*
+TODO
+username/login - czekamy na koncowke
+phone number (co z tym na backu) + prefix
+trzeba zrobic lepszy regex dla maila
+jakaś aktualizująca się max date / min date
+imie/nazwisko - regex na ąę?
+to samo z telefonem(czy mozna zrobic pozniej trim?)
+*/
+const validationSchema = yup.object({
+  firstName: yup
+    .string()
+    .matches(/^[a-zA-Z]+$/, "first name can only contain letters.")
+    .required("first name is required"),
 
-export default function Register({ updateStatus }: RegisterProps) {
-
-    //template of a function responsible for sending data of user being registered
-    const onSubmit = ({ user, link, validate }: onSubmitProps) => {
-        if (validate()) {
-            fetch(link, {
-                method: 'POST',
-                body: JSON.stringify({
-                    user
-                }),
-                headers: {
-                    'Content-type': 'application/json; charset=UTF-8',
-                },
-            })
-                .then((json) => console.log(json))
-
-            updateStatus();
-        }
-    }
+  lastName: yup
+    .string()
+    .matches(/^[a-zA-Z]+$/, "last name can only contain letters.")
+    .required("last name is required"),
+
+  email: yup.string().email("invalid e-mail").required("e-mail is required"),
+
+  diallingCode: yup
+    .string()
+    .matches(/^\+[0-9]{2}$/, "invalid dialling code")
+    .required("dialling code is required"),
+
+  phoneNumber: yup
+    .string()
+    .matches(/^[0-9]{9}$/, "invalid phone number")
+    .required("phone number is required"),
 
-    return (
-        <div className="Register">
-            <RegisterForm onSubmit={onSubmit} />
-        </div>
+  birthDate: yup
+    .date()
+    .required("birth date is required")
+    .min("1969-11-13", "Date is too early")
+    .max("2023-11-13", "Date is too late"),
+
+  password: yup
+    .string()
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/,
+      "must contain 8 characters, one uppercase, one lowercase, one number and one special character",
     )
-}
\ No newline at end of file
+    .required("password is required"),
+
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref("password"), ""], "passwords must match")
+    .required("password confirmation is required"),
+});
+
+export default function Register() {
+  const navigate = useNavigate();
+
+  const handleSubmit = async (
+    values: FormikValues,
+    { setSubmitting }: { setSubmitting: (isSubmitting: boolean) => void },
+  ) => {
+    try {
+      // Set submitting state to true to indicate form submission is in progress
+      setSubmitting(true);
+      // Send data to server
+      const response = await fetch(
+        "http://172.21.40.127:12038/auth/register-customer",
```

Comment:
not sure czy wpisywanie adresu na sztywno to dobry pomysł. Warto pomyśleć o jakimś utlis, do którego komponenty by się odwoływały. Ofc to bardziej uwaga na przyszłość